### PR TITLE
[R4R] improve the protocol details of Diff sync

### PR DESCRIPTION
**Pull request by guagualvcha**
### Description
This pr focus on the detailed improvement of diff sync.

### Rationale
The diff sync protocol does not behave well as we expected:
1. A peer gets less than 50% chance to diff sync.
2. A peer may receive duplicated diff layer.
3. The diff layer cache may pop recent difflayer.
4. Open too many files issue when enable `persist diff` 
5. The received difflayers maybe dropped once disconnected with the peer.

### Example
No

### Changes

No

### CODE REVIEW

**Review by FAvE01 — APPROVED** [review]
gh pr checkout 527

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -445,7 +445,7 @@ var (
 	DiffBlockFlag = cli.Uint64Flag{
 		Name:  "diffblock",
 		Usage: "The number of blocks should be persisted in db (default = 864000 )",
```
Comment (j75689) [inline cmd/utils/flags.go:447]: `default = 86400` right ?

File: eth/downloader/downloader.go [hunk 2]
```diff
@@ -232,27 +235,35 @@ type IPeerSet interface {
...
-			if ep := peers.GetDiffPeer(peerID); mode == FullSync && ep != nil {
-				hashes := make([]common.Hash, 0, len(headers))
-				for _, header := range headers {
-					hashes = append(hashes, header.Hash())
-				}
-				ep.RequestDiffLayers(hashes)
+		var hook = func(results []*fetchResult, stop chan struct{}) {
+			if dl.getMode() == FullSync {
+				go func() {
+					ticker := time.NewTicker(diffFetchTick * time.Millisecond)
```
Comment (j75689) [inline eth/downloader/downloader.go:241]: umm, is this correct? `10 * time.Millisecond * time.Millisecond` !?

**Review by cheersza05 — APPROVED** [review]

File: cmd/utils/flags.go (hunk 1 above)
Comment (guagualvcha) [reply cmd/utils/flags.go:447]: fixed

File: eth/downloader/downloader.go (hunk 2 above)
Comment (guagualvcha) [reply eth/downloader/downloader.go:241]: fixed

**Review by j75689 — APPROVED** [review]

**Review by yutianwu — APPROVED** [review]

**Pull request merged**